### ISSUE TITLE
Add detailed explanation of action helper & closure actions

### ISSUE THREAD

**Issue by bambery** [emberjs/guides#1719]
Recently I tweeted out an article by Miguel Camba @cibernox: "[Ember Closure Actions in Depth](http://miguelcamba.com/blog/2016/01/24/ember-closure-actions-in-depth/)" with a question as to whether this might be useful on the guides. @acorncom suggested I open an issue to take some of the content from this article for the guides. I've contacted and gotten the ok from Miguel to adapt the article for the guides.

The article was a very "aha" moment for myself regarding the purpose, value, and usage of actions, and I think there's a lot of value in the detailed breakdown that other people learning Ember could benefit from. 

I'd like to get some feedback/approval before I submit a PR, please.

**Comment by MarkMT** [comment]
I think that would be useful. FYI I've been looking at the overall presentation of actions in the guides and I think there are a few organizational issues that need a little attention. I'm just starting to work on this and hope to have something to share in a few days. It may be useful to take a look at what I produce to see if the two efforts will merge ok.

**Comment by MarkMT** [comment]
I have a branch with some work I've done in the last few days
https://github.com/MarkMT/guides/tree/actions_guide that I'm getting ready for a PR.

The changes are to the "Templates - Actions" guide: https://github.com/MarkMT/guides/blob/actions_guide/source/localizable/templates/actions.md

Among other things, this addresses my https://github.com/emberjs/guides/issues/1345 and the idea of nested actions that's discussed in the [API docs](http://emberjs.com/api/classes/Ember.Templates.helpers.html#toc_invoking-an-action) but not currently in the guide.

Happy to take any comments before I submit. I think there are also revisions needed to "Components - Handling Events" and "Components - Triggering Changes with Actions", which I hope to work on next.

**Comment by bambery** [comment]
I'll be glad to take a look at everything in this thread later tonight. I think this is a small but important enhancement to the existing docs.

**Comment by MarkMT** [comment]
Thanks. FYI I have done a little more work on templates/actions.md since yesterday. I think it's close to what I'd be happy to put in a PR.

**Comment by toddjordan** [comment]
@bambery I agree that's a great blog post.  Would you mind giving some analysis in this issue on what you think is clear from the post that have you that "aha moment" that is currently lacking in the guides?  We would love to hear your thoughts.

**Comment by theroncross** [comment]
That is a solid description. My overall impression on reading the guide as-is was that scrolling up and down to track the contents of each file was a pain and made it difficult to map. 
1. I would suggest we add a couple of 'checkpoints' that show the current state of the code in a single code box just to let everyone catch their breath and fill in gaps.
2. Perhaps some indication of 'core implementation' vs. 'advanced usage'.
